src/utils/github-utils.ts

```diff
@@ -0,0 +1,287 @@
+import { config } from "dotenv";
+import prompt from "prompt-sync";
+import {
+  deleteBranch,
+  getBranch,
+  getLatestCommit,
+  createBranch,
+  fetchReadme,
+  createBlob,
+  getCommitRef,
+  getBaseTree,
+  createTree,
+  createCommit,
+  updateRef,
+  createPullRequest,
+  mergePullRequest
+} from "../services/github-services";
+import { encodeToBase64, decodeBase64Content, logRed, logCyan, logPurple, logGreen, logOrange } from "./utils";
+import { createOrOverwriteFooter, shouldOverwriteFooter } from "./footer-utils";
+import { Repository, RepositoryStatus } from "../types/types";
+import { promptAndSaveFailedRepositories } from "./repository-error-file-utils";
+
+config();
+const updateBranchName = process.env.UPDATE_BRANCH_NAME;
+const promptSync = prompt();
+
+const repositoryStatuses: RepositoryStatus[] = [];
+
+/**
+ * Deletes a branch in a repository if it exists otherwise logs that no branch was found.
+ * 
+ * @param repositoryOBJ - An object containing the owner and repository name.
+ * @param repositoryOBJ.owner - The owner of the repository.
+ * @param repositoryOBJ.repository - The name of the repository.
+ * @param branchName - The name of the branch to delete.
+ * @returns A Promise that resolves with void.
+ * @throws Throws an error if there's an issue deleting the branch.
+ */
+const deleteBranchIfExists = async (repositoryOBJ: Repository, branchName: string): Promise<void> => {
+  const repoName = repositoryOBJ.repository
+
+  try {
+    await getBranch(repositoryOBJ, branchName);
```

Comment:
Should we check the existence of the branch before attempting to delete it?

Reply:
We are checking with the getBranch if the branch does not exist we get 404 error that is handled differently in the catch

Reply:
ok, fair enough, although it would be easier to read if this would be something like

```suggestion
    if (!await isExistingBranch(repositoryOBJ, branchName)) {
       console.log("Branch", logCyan(branchName), "does not exist in repository", logPurple(repoName), "Proceeding...");
       return;
    }
```

Reply:
I changed it to this